R/disease_progression.R

```diff
@@ -1,3 +1,79 @@
+#' @title Calculate recovery rates
+#' @description Calculates recovery rates for each individual in the population
+#' for storage in competing hazards object and future resolution
+#'
+#' @param variables the available human variables
+#' @param parameters model parameters
+#' @param recovery_outcome competing hazards object for recovery rates
+#' @noRd
+calculate_recovery_rates <- function(variables, parameters, recovery_outcome, dt_input){
+  
+  # Get correct input for dt depending on antimalarial resistance
+  if(isFALSE(parameters$antimalarial_resistance) & length(dt_input) == 1 & is.numeric(dt_input)){
+    dt_v <- dt_input
+  } else if (isTRUE(parameters$antimalarial_resistance)) {
+    dt_v <- dt_input$get_values(variables$state$get_index_of("Tr"))
+  }
+
+  recovery_rates <- numeric(length = parameters$human_population)
+  recovery_rates[variables$state$get_index_of("D")$to_vector()] <- 1/parameters$dd
+  recovery_rates[variables$state$get_index_of("A")$to_vector()] <- 1/parameters$da
+  recovery_rates[variables$state$get_index_of("U")$to_vector()] <- 1/parameters$du
+  recovery_rates[variables$state$get_index_of("Tr")$to_vector()] <- 1/dt_v
+  recovery_outcome$set_rates(recovery_rates)
```

Comment:
It's perhaps more efficient to update variable called `recovery_rates` in the progression processes. Then all of this becomes.

```suggestion
  recovery_outcome$set_rates(variables$recovery_rates$get_values())
```

Reply:
I think this actually tidies up some of the issues I was having integrating with @tbreweric 's dt variable. I'm guessing that if we're storing all the recovery rates as a variable then we can replace the dt variable with a recovery_rates variable and still store all the relevant information. Let me know what you think with what I've done.